File: app/src/main/java/com/gunnys/eundunhealth/domain/model/Goal.kt
package com.gunnys.eundunhealth.domain.model

import androidx.compose.runtime.Immutable
import java.time.Instant

enum class GoalType(val key: String, val unit: String, val label: String) {
    WEIGHT("weight", "kg", "목표 체중"),
    BODY_FAT("body_fat", "%", "목표 체지방률"),
    ;

    companion object {
        fun fromKey(key: String): GoalType? = entries.firstOrNull { it.key == key }
    }
}

@Immutable
data class Goal(
    val type: GoalType,
    val targetValue: Float,
    val createdAt: Instant?,
)

@Immutable
data class ProfileHistoryPoint(
    val heightCm: Float,
    val weightKg: Float,
    val bodyFatPct: Float?,
    val muscleMassKg: Float?,
    val recordedAt: Instant?,
)

File: backend/app/schemas/goal.py
from typing import Annotated, Literal

from pydantic import Field

from app.schemas.base import CamelSchema


class GoalRequest(CamelSchema):
    """PUT /goals — goal_type별 1개. 동일 type 재등록 시 target_value 갱신."""

    goal_type: Literal["weight", "body_fat"]
    target_value: Annotated[float, Field(gt=0, le=500)]


class GoalResponse(CamelSchema):
    goal_type: str
    target_value: float
    created_at: str


class ProfileHistoryEntry(CamelSchema):
    height_cm: float
    weight_kg: float
    body_fat_pct: float | None
    muscle_mass_kg: float | None
    recorded_at: str
